src/DynamicData/Cache/ObservableCache.cs

```diff
@@ -240,4 +322,103 @@ private void InvokePreview(ChangeSet<TObject, TKey> changes)
             }
         }
     }
+
+    private void InvokeCountNext()
+    {
+        lock (_locker)
+        {
+            if (_countChanged.IsValueCreated)
+            {
+                _countChanged.Value.OnNext(_readerWriter.Count);
+            }
+        }
+    }
+
+    private void ResumeCount()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Count without Suspend Count instance");
+            _suspensionTracker.Value.ResumeCount();
+        }
+    }
+
+    private void ResumeNotifications()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Notifications without Suspend Count instance");
+            _suspensionTracker.Value.ResumeNotifications();
+        }
+    }
+
+    private class SuspensionTracker(Action<ChangeSet<TObject, TKey>> onResumeNotifications, Action onResumeCount) : IDisposable
+    {
+        private readonly BehaviorSubject<bool> _notifySuspendSubject = new(false);
+
+        private readonly Action<ChangeSet<TObject, TKey>> _onResumeNotifications = onResumeNotifications;
+
+        private readonly Action _onResumeCount = onResumeCount;
+
+        private IEnumerable<Change<TObject, TKey>> _pending = Enumerable.Empty<Change<TObject, TKey>>();
+
+        private int _countSuspendCount;
+
+        private int _notifySuspendCount;
+
+        public bool CountSuspended => _countSuspendCount > 0;
+
+        public bool NotificationsSuspended => _notifySuspendCount > 0;
+
+        public IObservable<bool> NotificationsSuspendedStatus => _notifySuspendSubject;
+
+        public void AddPending(IEnumerable<Change<TObject, TKey>> changes)
+        {
+            Debug.Assert(changes is not null, "Don't pass in a null Enumerable");
+            Debug.Assert(NotificationsSuspended, "Shouldn't be adding pending values if notifications aren't suspended");
+            _pending = _pending.Concat(changes);
+        }
+
+        public void SuspendNotifications()
+        {
+            if (++_notifySuspendCount == 1)
+            {
+                Debug.Assert(!_pending.Any(), "Shouldn't be any pending values if suspend was just started");
+                Debug.Assert(!_notifySuspendSubject.Value, "SuspendSubject should be false for the first suspend call");
+                _notifySuspendSubject.OnNext(true);
+            }
+        }
+
+        public void SuspendCount() => ++_countSuspendCount;
+
+        public void ResumeNotifications()
+        {
+            if (--_notifySuspendCount == 0 && !_notifySuspendSubject.IsDisposed)
+            {
+                // Fire pending changes to existing subscribers
+                if (_pending.Any())
```

Comment:
Swap to just `.Count` if you swap `_pending` to a `List<T>` as suggested above.

Reply:
Yeah, I don't like `Any` but there's little choice with enumerables.  Let me know what you think about avoiding the copy.

Reply:
I just went with List.  It's clearer.